back.py

```diff
@@ -78,3 +80,34 @@ def get_table_content(self, db_name, exp_names, columns):
         cursor = collection.find(filter=filter, projection=projection)
 
         return generate_experiment_table_rows(cursor, columns)
+
+    def filter_rows_by_query(self, rows, query):
+        '''
+        Returns rows that are filtered by the query.
+        Uses the pandas.DataFrame.query method
+        '''
+
+        if query.strip() == '':
+            return rows
+
+        df = DataFrame(rows)
+
+        # temporarily replacing '.' in columns to '__'
+        # for pandas syntax reasons
+        normalize_col_names = lambda x: x.replace('.', '__')
+        df = df.rename(columns=normalize_col_names)
+
+        # filter rows with query
+        try:
+            df_filtered = df.query(normalize_col_names(query), locals={})
+        except:
+            logger.warning(f'Error in query "{query}"')
```

Comment:
Maybe indicate the specific error in the logging message here, that could help debugging queries.

Reply:
> We should also find a way to prevent bugs in edge cases of the x.replace('.', '__') trick, for example columns named bla_.bla will be renamed bla._bla.

Yes, currently, we obviously do not support all column names (but nor does pandas evidently...). What could be done, is to select a symbol that works for pandas, and that is not contained in the query. But what happens when all symbols are used in the query?